li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -552,17 +589,48 @@ synchronized public long position(TopicPartition partition, Duration timeout) {
 
   @Override
   synchronized public OffsetAndMetadata committed(TopicPartition partition) {
-    throw new UnsupportedOperationException("Not implemented yet");
+
+    return committed(partition, _defaultApiTimeout);
   }
 
   @Override
   synchronized public OffsetAndMetadata committed(TopicPartition partition, Duration timeout) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    if (partition == null) {
+      throw new IllegalArgumentException("partition cannot be null");
+    }
+
+    long now = System.currentTimeMillis();
+    long deadlineTimeMs = now + timeout.toMillis();
+    while (now < deadlineTimeMs) {
+      LiKafkaConsumer<K, V> consumer = getPerClusterConsumerForTopic(partition.topic());
+      if (consumer != null) {
+        return consumer.committed(partition, timeout);
+      }
+
+      try {
+        Thread.sleep(Math.min(deadlineTimeMs - System.currentTimeMillis(), _retryBackoffMs));
```

Comment:
Can _deadlineTimeMs - System.currentTimeMillis()_ be negative and cause infinite sleep?

Reply:
sleep would throw for a negative